src/graph/OrderByExecutor.cpp

```diff
@@ -0,0 +1,226 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "base/Base.h"
+#include "graph/OrderByExecutor.h"
+
+namespace nebula {
+namespace graph {
+namespace cpp2 {
+
+bool ColumnValue::operator < (const ColumnValue& rhs) const {
+    DCHECK_EQ(type_, rhs.type_);
+    auto& lhs = *this;
+    switch (lhs.type_) {
+        case Type::bool_val:
+        {
+            if (!(lhs.value_.bool_val == rhs.value_.bool_val)) {
+                return lhs.value_.bool_val < rhs.value_.bool_val;
+            }
+            break;
+        }
+        case Type::integer:
+        {
+            if (!(lhs.value_.integer == rhs.value_.integer)) {
+                return lhs.value_.integer < rhs.value_.integer;
+            }
+            break;
+        }
+        case Type::id:
+        {
+            if (!(lhs.value_.id == rhs.value_.id)) {
+                return lhs.value_.id < rhs.value_.id;
+            }
+            break;
+        }
+        case Type::single_precision:
+        {
+            if (!(lhs.value_.single_precision == rhs.value_.single_precision)) {
+                return lhs.value_.single_precision < rhs.value_.single_precision;
+            }
+            break;
+        }
+        case Type::double_precision:
+        {
+            if (!(lhs.value_.double_precision == rhs.value_.double_precision)) {
+                return lhs.value_.double_precision < rhs.value_.double_precision;
+            }
+            break;
+        }
+        case Type::str:
+        {
+            if (!(lhs.value_.str == rhs.value_.str)) {
+                return lhs.value_.str < rhs.value_.str;
+            }
+            break;
+        }
+        case Type::timestamp:
+        {
+            if (!(lhs.value_.timestamp == rhs.value_.timestamp)) {
+                return lhs.value_.timestamp < rhs.value_.timestamp;
+            }
+            break;
+        }
+        case Type::year:
+        {
+            if (!(lhs.value_.year == rhs.value_.year)) {
+                return lhs.value_.year < rhs.value_.year;
+            }
+            break;
+        }
+        case Type::month:
+        {
+            if (!(lhs.value_.month == rhs.value_.month)) {
+                return lhs.value_.month < rhs.value_.month;
+            }
+            break;
+        }
+        case Type::date:
+        {
+            if (!(lhs.value_.date == rhs.value_.date)) {
+                return lhs.value_.date < rhs.value_.date;
+            }
+            break;
+        }
```

Comment:
To date, datetime, are you comparing by string or by corresponding value?

Reply:
by value, how do u think? it depends on the implementation of storage actually. Date and Datetime is not implement in storage. For now, i am doing the sort with the union type of ColumnValue, because it is comparable.

Reply:
Although `date` and `datetime` is not implemented,  what I really care about is how to store these two types, whether they are stored as strings or converted to timestamps.

Reply:
Do u mean that _how to store in storage_?